### PR TITLE
DAOS-4121 control: fix pool create size with raw nvme bytes

### PR DESCRIPTION
**Pull request by tanabarr**
Due to changes made to represent NVMe SSD capacity in base-10,
raw bytes provided to dmg pool create -n parameter will be converted
to a slightly lower value. Fix issue by detecting and handling unit
size specifiers individually.

Signed-off-by: Tom Nabarro <tom.nabarro@intel.com>

### CODE REVIEW

File: src/control/cmd/dmg/pool.go [hunk 1]
```diff
@@ -179,16 +207,15 @@ func calcStorage(log logging.Logger, scmSize string, nvmeSize string) (
 		return
 	}
 
-	nvmeBytes, err = getSize(nvmeSize)
+	// NVMe/SSD storage specified in MB (base 10), not Mib (base 2)
```
[the remark this answers is not in the document]
Comment (tanabarr) [reply src/control/cmd/dmg/pool.go]: yes , that's correct. documentation has been updated in https://github.com/daos-stack/daos/pull/1833

[the remark this answers is not in the document]
Comment (tanabarr) [reply src/control/cmd/dmg/pool.go]: will update the help text for the command

**Comment by tanabarr** [comment]
> Basically looks correct, a couple minor comments. I'm starting to think there might be some value in writing our own size conversion library at this point, if none of the existing ones do what we want (interpreting base 10 as well as base 2 units). It seems like it might be simpler to do all of it ourselves.

Yes the ticket for that is https://jira.hpdd.intel.com/browse/DAOS-4049 which Mike has decided to commandeer, for that I was switching to go-humanize package which can parse and represent both base-2/10. that PR was almost complete but I was told not to land it until after 1.0 (fix version on the ticket is 1.2) as it was too big a change to land as a simple fix between 0.9 and 1.0. Therefore this PR was an interim fix to make sure the issues Phil raised (pool NVMe size incorrect when specifying raw bytes in pool create) get fixed by 1.0.

**Comment by tanabarr** [comment]
https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-1832/3/execution/node/514/log functional_hardware tests failing on two intermittent failures:
- IorSmall "failed to start server in format mode" https://jira.hpdd.intel.com/browse/DAOS-4085
- MdtestSmall "failed to start server in normal mode" https://jira.hpdd.intel.com/browse/DAOS-4059 / https://jira.hpdd.intel.com/browse/DAOS-4085

**Comment by mjmac** [comment]
@tanabarr: Given the number of issues we've had related to this logic, I think it would be prudent to consider just switching over to go-humanize rather than adding more workarounds which will be removed right away. 0.9 has already been tagged, so I think we can make a case for switching to the new library as a solution for several pre-1.0 bugs. If you have a PR that is close to production-ready, please finish it and push for review.

**Comment by tanabarr** [comment]
replaced by https://github.com/daos-stack/daos/pull/1898

**Pull request closed without merging**